src/util/bip32.rs

```diff
@@ -61,15 +226,17 @@ pub struct ExtendedPrivKey {
     /// Private key
     pub private_key: PrivateKey,
     /// Chain code
-    pub chain_code: ChainCode
+    pub chain_code: ChainCode,
+    _marker: ::std::marker::PhantomData<R>
 }
-serde_string_impl!(ExtendedPrivKey, "a BIP-32 extended private key");
+serde_string_impl!(ExtendedPrivKey<R: VersionResolver>, "a BIP-32 extended private key");
 
 /// Extended public key
 #[derive(Copy, Clone, PartialEq, Eq, Debug, PartialOrd, Ord, Hash)]
-pub struct ExtendedPubKey {
-    /// The network this key is to be used on
-    pub network: Network,
+pub struct ExtendedPubKey<R: VersionResolver> {
```

Comment:
I'm not a fan of the fact that this type becomes generic. I think it must be possible to have this functionality without this type being generic.